src/components/Reactions/hooks.ts

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+
+import {useStableCallback} from '../utils/useStableCallback';
+
+import type {ReactionProps} from './Reaction';
+import {useReactionsContext} from './context';
+
+const DELAY = {
+    focusTimeout: 600,
+    openTimeout: 200,
+    closeTimeout: 100,
+} as const;
+
+export function useReactionsPopup(
+    reaction: ReactionProps,
+    ref: React.RefObject<HTMLButtonElement>,
+) {
+    const {value} = reaction;
+
+    const {openedTooltip: currentHoveredReaction, setOpenedTooltip: setCurrentHoveredReaction} =
+        useReactionsContext();
+
+    const {delayedCall: setDelayedOpen, clearTimeoutRef: clearOpenTimeout} = useTimeoutRef();
+    const {delayedCall: setDelayedClose, clearTimeoutRef: clearCloseTimeout} = useTimeoutRef();
+
+    const open = useStableCallback(() => {
+        setCurrentHoveredReaction({reaction, open: true, ref});
+    });
+
+    const close = useStableCallback(() => {
```

Comment:
As Alexey mentioned above we can just use useCallback instead. Also the implementation of `useStableCallback` violates React's rules (writing or reading refs during render)

Reply:
Removed `useStableCallback` completely